src/device_uuid.h

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <array>
+#include <stdexcept>
+#include <cstdint>
+#include <string>
+#include <vulkan/vulkan.hpp>
+
+
+template<std::size_t Size>
+constexpr std::array<char, 2 * Size> decode_UUID(const std::array<uint8_t, Size>& bytes)
```

Comment:
It also does not cost anything else than microseconds in compilation time. To confess, i think that algorithm layer should be as general as it is reasonable. I also did not want to couple algorithm with vulkan internals. But if you really want me to change it i will do it

Reply:
It's indeed a small issue, but why increase complexity even a little to support a use case that's unlikely to ever be required (i.e., size != VK_UUID_SIZE, especially since we can trivially add this back if we ever need the flexibility.